### PR TITLE
Add appsync resource

### PR DESCRIPTION
**Pull request by cahn1**
This PR is to add [AppSync](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/appsync.html) `graphql-api` resource.

Also adding support for local and Firewall Manager deployed WebACLs with regex support. This is subsequent PR to support `graphql-api` from [pr/7706](https://github.com/cloud-custodian/cloud-custodian/pull/7706).

With this PR, we can use regex expression and it still works with local WAF as well.

```
    policies:
      - name: filter-graphql-api-wafv2
        resource: graphql-api
        filters:
          - type: wafv2-enabled
            state: false
            web-acl: test-waf-v2

      - name: filter-graphql-api-wafv2-regex
        resource: graphql-api
        filters:
          - type: wafv2-enabled
            state: false
            web-acl: .*FMManagedWebACLV2-?FMS-.*
```

Similar filter and action were used from [pr/7706](https://github.com/cloud-custodian/cloud-custodian/pull/7706).

Complete example policy:
```
    policies:
      - name: set-wafv2-for-graphql-api-regex
        resource: graphql-api
        filters:
          - type: wafv2-enabled
            state: false
            web-acl: .*FMManagedWebACLV2-?FMS-.*
        actions:
          - type: set-wafv2
            state: true
            force: true
            web-acl: FMManagedWebACLV2-?FMS-TestWebACL
```

### CODE REVIEW

File: c7n/resources/appsync.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                  - type: wafv2-enabled
+                    state: false
+                    web-acl: .*FMManagedWebACLV2-?FMS-.*
+                actions:
+                  - type: set-wafv2
+                    state: true
+                    force: true
+                    web-acl: FMManagedWebACLV2-?FMS-TestWebACL
+    """
+    permissions = ('appsync:UpdateGraphqlApi', 'wafv2:ListWebACLs')
```
Comment (kapilt) [inline c7n/resources/appsync.py:124]: missing permissions on wafv2 for associate/dissasociate afaics

Comment (cahn1) [reply c7n/resources/appsync.py]: @kapilt oops, permissions updated.

File: tests/test_appsync.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                "filters": [{"type": "wafv2-enabled", "state": False,
+                             "web-acl": ".*FMManagedWebACLV2-?FMS-.*"}],
+                "actions": [{"type": "set-wafv2", "state": True,
+                             "web-acl": ".*FMManagedWebACLV2-?FMS-.*"}]
+            },
+            session_factory=factory,
+        )
+        with self.assertRaises(ValueError) as ctx:
+            p.run()
+            self.assertTrue('matching to none or multiple webacls' in str(
```
Comment (kapilt) [inline tests/test_appsync.py:117]: syntactically this construction is odd, this line shouldn't be reached, if it does, ctx.exception would be none afaics.

**Review by kapilt — APPROVED** [review]
lgtm, thanks.. @harishachappa did you want to have a look?

**Comment by harishachappa** [comment]
lgtm, thanks

File: tests/test_appsync.py (hunk 2 above)
Comment (cahn1) [reply tests/test_appsync.py]: @kapilt Yes, you are correct. It wasn't reachable. out-dented.

**Pull request merged**